### PR TITLE
Issue 25

### PR DESCRIPTION
**Pull request by rendel**

### CODE REVIEW

File: src/main/java/solutions/siren/join/action/coordinate/FilterJoinVisitor.java [hunk 1]
```diff
@@ -209,32 +195,83 @@ private void checkForFailure(FilterJoinNode node) {
...
+    protected TermsByQueryRequest getTermsByQueryRequest(FilterJoinNode node) {
+      String[] lookupIndices = node.getLookupIndices();
+      String[] lookupTypes = node.getLookupTypes();
+      String lookupPath = node.getLookupPath();
+      XContentBuilder lookupQuery = node.getLookupQuery();
+      TermsByQueryRequest.Ordering ordering = node.getOrderBy();
+      Integer maxTermsPerShard = node.getMaxTermsPerShard();
+      TermsByQueryRequest.TermsEncoding termsEncoding = node.getTermsEncoding();
+
+      return new TermsByQueryRequest(lookupIndices).field(lookupPath)
```
Comment (scampi) [inline src/main/java/solutions/siren/join/action/coordinate/FilterJoinVisitor.java]: nitpick: put `.field(lookupPath)` on the next line so it is aligned with the rest.

File: src/main/java/solutions/siren/join/action/terms/TermsByQueryRequest.java [hunk 2]
```diff
@@ -285,7 +324,7 @@ public void writeTo(StreamOutput out) throws IOException {
       out.writeBoolean(false);
     } else {
       out.writeBoolean(true);
-      out.writeString(ordering.name());
+      out.writeVInt(ordering.ordinal());
```
Comment (scampi) [inline src/main/java/solutions/siren/join/action/terms/TermsByQueryRequest.java]: :+1:

File: src/main/java/solutions/siren/join/index/query/FieldDataTermsQueryHelper.java [hunk 3]
```diff
@@ -73,28 +52,73 @@ public final static long hash(BytesRef term) {
   /**
    * Encodes a long into the byte array dst at the given offset.
    */
-  public final static void writeInt(byte[] dst, int offset, int i) {
-    dst[offset] = ((byte) (i >> 24));
-    dst[offset + 1] = ((byte) (i >> 16));
-    dst[offset + 2] = ((byte) (i >> 8));
-    dst[offset + 3] = ((byte) i);
+  public final static void writeInt(BytesRef dst, int i) {
```
Comment (scampi) [inline src/main/java/solutions/siren/join/index/query/FieldDataTermsQueryHelper.java]: I don't how many times this is called, but revert to previous `writeInt` and add `dst.offset += 4;` at the end ?

**Comment by scampi** [comment]
nit-picking only, LGTM

File: src/main/java/solutions/siren/join/action/coordinate/FilterJoinVisitor.java (hunk 1 above)
Comment (rendel) [inline src/main/java/solutions/siren/join/action/coordinate/FilterJoinVisitor.java]: ok

File: src/main/java/solutions/siren/join/index/query/FieldDataTermsQueryHelper.java (hunk 3 above)
Comment (rendel) [inline src/main/java/solutions/siren/join/index/query/FieldDataTermsQueryHelper.java]: ok

**Pull request merged**